addOns/pscanrulesAlpha/src/main/java/org/zaproxy/zap/extension/pscanrulesAlpha/SubResourceIntegrityAttributeScanRule.java

```diff
@@ -99,23 +111,63 @@ public static boolean contains(String tag) {
     @Override
     public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
 
+        SiteMap tree = Model.getSingleton().getSession().getSiteTree();
         List<Element> sourceElements = source.getAllElements();
         sourceElements.stream()
                 .filter(element -> SupportedElements.contains(element.getName()))
                 .filter(unsafeSubResource(msg.getRequestHeader().getHostName()))
                 .forEach(
-                        element ->
-                                newAlert()
-                                        .setRisk(Alert.RISK_MEDIUM)
-                                        .setConfidence(Alert.CONFIDENCE_HIGH)
-                                        .setDescription(getString("desc"))
-                                        .setSolution(getString("soln"))
-                                        .setReference(getString("refs"))
-                                        .setEvidence(element.toString())
-                                        .setCweId(345) // CWE-345: Insufficient Verification of Data
-                                        // Authenticity
-                                        .setWascId(15) // Application Misconfiguration
-                                        .raise());
+                        element -> {
+                            String integrityHash = calculateIntegrityHash(msg, element, tree);
+                            String otherInfo =
+                                    integrityHash.isEmpty()
+                                            ? ""
+                                            : Constant.messages.getString(
+                                                    MESSAGE_PREFIX + "otherInfo", integrityHash);
+                            newAlert()
+                                    .setRisk(Alert.RISK_MEDIUM)
+                                    .setConfidence(Alert.CONFIDENCE_HIGH)
+                                    .setDescription(getString("desc"))
+                                    .setSolution(getString("soln"))
+                                    .setReference(getString("refs"))
+                                    .setEvidence(element.toString())
+                                    .setCweId(345) // CWE-345: Insufficient Verification of Data
+                                    // Authenticity
+                                    .setWascId(15) // Application Misconfiguration
+                                    .setOtherInfo(otherInfo)
+                                    .raise();
+                        });
+    }
+
+    private String calculateIntegrityHash(HttpMessage msg, Element element, SiteMap tree) {
+        String src = element.getAttributeValue("src");
+        String integrityHash = null;
```

Comment:
This should be set as empty string so that the ternary statement for setting the otherInfo works properly.

`         String integrityHash = "";`

Reply:
Oh yea .. I overlooked that. But what is better, using an empty string or null?

Reply:
But that hash can never be an empty string unless there isn't any hash present. So I guess in this case it doesn't matter.

Reply:
For my two cents using empty string is better than null